homotopy-model/src/history.rs

```diff
@@ -173,6 +173,22 @@ impl History {
         actions
     }
 
+    pub fn get_last_import_segment(&self) -> Vec<super::proof::Action> {
+        let mut actions = Vec::new();
+        for a in self
+            .snapshots
+            .ancestors_of(self.current)
+            .filter_map(|n| self.snapshots.with(n, |s| s.action.clone()).flatten())
+        {
+            actions.push(a.clone());
+            if matches!(a, super::proof::Action::ImportProof(_)) {
```

Comment:
You can use `Iterator::take_while` to make this neater.

Reply:
I'm not sure I can. I need to include the last ImportProof. I don't think there is a neat way of expressing it.